### ISSUE TITLE
restarts in docs don't work sometimes

### ISSUE THREAD

**Issue by erg** [factor/factor#2310]
Refreshing this and selecting ``USE: ui.tools.inspector`` keeps growing the restart list with the same option.

```
USING: help.markup help.syntax ui.commands
ui.gadgets.panes ui.gadgets.editors ui.tools.inspector.slots kernel ;
IN: ui.tools.inspector+docs

ARTICLE: "ui-inspector-edit" "Editing slot values in the inspector"
"Slot values can be edited in the inspector. Clicking the ellipsis to the left of the slot's textual representation displays a slot editor gadget. A text representation of the object can be edited in the slot editor. The parser is used to turn the text representation back into an object. Keep in mind that some structure is lost in the conversion; see " { $link "prettyprint-limitations" } "."
$nl
"The slot editor's text editing commands are standard; see " { $link "ui.gadgets.editors" } "."
$nl
"The slot editor has a toolbar containing various commands."
{ $command-map slot-editor "toolbar" } ;

ARTICLE: "ui-inspector" "UI inspector"
"The graphical inspector provides functionality similar to the terminal inspector (see " { $link "inspector" } "), adding in-place editing of slot values."
$nl
"To display an object in the UI inspector, right-click a presentation and choose " { $strong "Inspector" } " from the menu that appears. The inspector can also be opened from the listener using a word:"
{ $subsections inspector }
"Objects may extend the inspector window with a content gadget, in order to provide a rich representation of the object's content in the inspector."
{ $subsections content-gadget }
"The inspector embeds a table gadget, which supports keyboard navigation; see " { $link "ui.gadgets.tables" } ". It also provides a few other commands:"
{ $command-map inspector-gadget "toolbar" }
{ $command-map inspector-gadget "multi-touch" }
"The UI inspector is an instance of " { $link inspector-gadget } "."
{ $subsections "ui-inspector-edit" } ;

HELP: inspector
{ $values { "obj" object } }
{ $description "Opens a new inspector window displaying the slots of " { $snippet "obj" } "." } ;

ABOUT: "ui-inspector"
```

![IMG_20200613_231746_exported_256026036902536455](https://user-images.githubusercontent.com/27430/84584659-99978880-adcc-11ea-89e3-949fe2dca81e.jpg)

**Comment by erg** [comment]
Looks like ``compute-restarts`` recursively calls itself N times, adding one per restart.

```
\ debugger-popup see

M: condition compute-restarts
    [ error>> compute-restarts ] [
        [ restarts>> ] [ continuation>> [ <restart> ] curry ] bi
        { } assoc>map
    ] bi append ;

USING: accessors continuations lexer ;
M: lexer-error compute-restarts error>> compute-restarts ;

USING: continuations kernel ;
M: object compute-restarts drop { } ;

USING: accessors continuations source-files.errors ;
M: source-file-error compute-restarts error>> compute-restarts ;
```


```
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~source-file-error~ ~array~ ~continuation~ }
--- Entering M\ condition compute-restarts
error T{ condition f ~no-word-error~ ~array~ ~continuation~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ }
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Leaving M\ condition compute-restarts
seq { ~restart~ ~restart~ ~restart~ ~restart~ ~restart~ ~restart~...
--- Entering <restart-list>
debugger T{ debugger f ~array~ ~array~ f f f ~array~ f t f f f ~array~...
--- Leaving <restart-list>
gadget T{ table f ~array~ ~array~ f f f ~array~ f t f f f ~array~...
```

**Comment by erg** [comment]
more minimal

```factor
IN: ui.tools.inspector+lol

\ inspector drop
```

**Comment by mrjbq7** [comment]
Fixed in ff25e303794ff1efa63487b84dae59e2dc50ee56.
